lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -62,75 +66,169 @@ def __init__(self, config):
         self.excluded_features = ('source', 'exon', 'fasta_record')
         self.ont_mappings = load_ontology_mappings('/kb/module/data')
         self.code_table = 11
-        self.re_api_url = config.re_api_url
         # dict with feature 'id's that have been used more than once.
         self.used_twice_identifiers = {}
+
+        # related info for genome process and upload
+        self.genome_name = None
+        self.genome_data = None
+        self.genome_meta = None
+        self.genome_type = None
+        self.consolidated_file = None
+        self.input_directory = None
+        self.file_handle = None
```

Comment:
Is this actually a file handle? If so I'd make the code that needs it manage it, otherwise it's highly likely they'll leak

Reply:
This is what dfu.file_to_shock_mass return. So technically is not a file handle but a dict that named `file_handle`?

Reply:
Ok, so a handle service handle. That's fine. Maybe consider renaming it so it's a bit less scary

Reply:
I will rename it to `shock_output`

Reply:
Shock is dead, long live blobstore

Reply:
`blobstore_output` then

Reply:
Come to think of it it might be the handle service output. That's more likely

Reply:
👍

Reply:
so looking at https://github.com/kbaseapps/DataFileUtil/blob/master/DataFileUtil.spec#L136-L141, the returned structure is a mix of data from blobstore and the handle service. I don't think this is really important enough to rename again unless you want to; I just don't want people to think it's an actual open file handle

Reply:
:+1:

Reply:
I assume minio is behind blobstore? What exactly is handle service?

Reply:
1. Right
2. https://github.com/kbase/handle_service2/. Its current role is to help manage ACL synchronization between the workspace and the blobstore